### PR TITLE
RNA buttons: add styles and storybook documentation

### PR DESCRIPTION
**Pull request by keoshi**
#### Changes proposed in this Pull Request:

* Add color variables.
* Add overriding styles for the `@wordpress/components` buttons.
* Added a documentation page in Storybook.

#### Jetpack product discussion

p6TEKc-5rF-p2

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions for Storybook:

* Go to the Storybook folder `cd projects/js-packages/storybook`
* Run `pnpm install`
* Run `pnpm run storybook:dev`
* Storybook should open in your browser.
* Go to Components → Buttons → Docs and play around with the various buttons.

#### Screenshot

image

### CODE REVIEW

**Review by madebynoam — DISMISSED** [review]

**Review by fgiannar — COMMENTED** [review]
Wonderful work, Filipe!

I left a few inline comments :)

Out of curiosity, I see you are using the MDX format, I guess because you wish to add custom docs, right? We could take it one step further maybe by utlizing `Templates` and `Controls` that will allow us, eg, to edit a button's label but I think this is totally optional in this case.

**Comment by keoshi** [comment]
@sergeymitr Thanks for the feedback and glad to hear this is helpful! I don't mind you jumping in one bit, please do it whenever you need.

> That made the button background turn black, as any other primary button according to your styling. Should we keep it black, or keep the blue background?

The RNA/Emerald standard going forward is going to be black buttons, at least until something major changes.

Will take a look at disabled buttons too!

**Comment by keoshi** [comment]
Added disabled styles and buttons to the storybook page, here's how it looks like:

image

File: projects/js-packages/base-styles/_buttons.scss [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+	&:disabled {
+		color: rgba( var( --jp-color-white-rgb ), 0.4 );
+	}
+
+	&.is-secondary {
+
+		&:focus:not(:disabled) {
+			color: var( --jp-color-red-50 );
+		}
```
Comment (sergeymitr) [inline projects/js-packages/base-styles/_buttons.scss:80]: One minor issue I noticed.
The button label becomes unreadable on hover, should we make it white?
Screen Shot 2021-11-03 at 3 44 38 PM

```suggestion
		}
		
		&:disabled {
			color: var( --jp-color-white );
		}
```

File: projects/js-packages/base-styles/_buttons.scss [hunk 2]
```diff
@@ -0,0 +1,93 @@
+@import './variables';
+@import './colors';
+
+.components-button {
```
Comment (sergeymitr) [inline projects/js-packages/base-styles/_buttons.scss:4]: We discussed it with the team today, and discovered that these styles will also apply to non-RNA buttons, and in some cases could affect even non-Jetpack elements on the page.
We're discussing possible solutions in Slack: p1635967176436400-slack-CBG1CP4EN

File: projects/js-packages/connection/components/connect-screen/layout/style.scss [hunk 3]
```diff
@@ -86,8 +86,18 @@
...
+					color: #000;
+					text-decoration: underline;
+				}
 			}
 		}
+
+		a.jp-connection-screen-icon {
+			text-decoration: none;
+			padding-left: 5px;
+		}
```
Comment (sergeymitr) [inline projects/js-packages/connection/components/connect-screen/layout/style.scss:100]: Reintroducing these lines, were lost in transition when we merged #21521.
They are responsible for styling the "See all Jetpack features" button and the "external" icon:

Screen Shot 2021-11-04 at 10 44 06 AM

File: projects/js-packages/connection/components/connect-screen/layout/style.scss [hunk 4]
```diff
@@ -86,8 +86,18 @@
 				padding-left: 30px;
 				margin-bottom: 9px;
 				color: var( --jp-black );
+
+				a {
```
Comment (fgiannar) [inline projects/js-packages/connection/components/connect-screen/layout/style.scss:90]: @sergeymitr Do we need to repeat the link styles in `li`? They are also defined in line 60

Comment (sergeymitr) [reply projects/js-packages/connection/components/connect-screen/layout/style.scss:90]: It seems like we do. I tried without it, and there was no underline.
Apparently some other style overwrites it.
I didn't dig deep into it though ¯\\\_(ツ)\_/¯

Comment (fgiannar) [reply projects/js-packages/connection/components/connect-screen/layout/style.scss:90]: Ok I will look into this more while working on the base styles

**Comment by fgiannar** [comment]
Hi @keoshi ,

While testing this I notices that we don't have buttons styles for `active` button state.

This results in:
Screenshot 2021-11-05 at 12 46 47 PM

Should we add those as well?

**Comment by fgiannar** [comment]
@keoshi I'm thinking to use this PR as the basis for the work needed to properly configure the base styles package.
While working on it I can address the issues we've mentioned so I can take the burden of having to maintain / update the PR yourself.
Just let me know here in a comment the changes you'd wish to add and I'd be happy to incorporate them :)

**Comment by keoshi** [comment]
Sorry @fgiannar — I missed the latest notification on this PR somehow.

The only thing I believe is missing are the active states. Do you need those to be specified beyond what we currently have in Figma?

Thank you, appreciate you taking this through the finish line!!

**Comment by sergeymitr** [comment]
Hi @keoshi.
Just a quick update on the state of things.
We're still designing the Base-Styles package, considering different approaches to make it easy to use and prevent it from disrupting other styles on the page.
Here's the discussion we're having: p9dueE-3Tq-p2

**Comment by samiff** [comment]
@sergeymitr Is the base-styles package still being discussed, or has it reached a more stable status?

**Comment by kraftbj** [comment]
Howdy y'all — what is the status of this one? I'm going to mark it as "needs author reply" to clean up the "Needs Review" board since there are multiple conflicts and needs a master-merge.

**Comment by renatoagds** [comment]
Closing in favor of #23584

**Pull request closed without merging**